js/fileactions.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2014
+ *
+ * This file is licensed under the Affero General Public License version 3
+ * or later.
+ *
+ * See the COPYING-README file.
+ *
+ */
+window.addEventListener('DOMContentLoaded', () => {
```

Comment:
I think we can simplify this, since we're registering the file action and using the menu observer to hide or show it, we don't need any other logic here than actually registering the action itself. So I believe we would only need from line 13 to 50. We of course still need the event listener for "DOMContentLoaded" around it.

Reply:
Done and tested okay

Reply:
I believe we can simplify it a bit more than what has been done here, but let's cover it on the next meeting